TSS.NET/TSS.Net.UWP/CryptoAsym.cs

```diff
@@ -0,0 +1,1010 @@
+        {
+            var cs = new AsymCryptoSystem();
+
+            TpmAlgId keyAlgId = pubKey.type;
+            cs.PublicParms = pubKey.Copy();
+
+            // Create an algorithm provider from the provided PubKey
+            switch (keyAlgId)
+            {
+                case TpmAlgId.Rsa:
+                {
+                    RawRsa rr = null;
+                    byte[] prime1 = null,
+                            prime2 = null;
+                    if (privKey != null)
+                    {
+                        rr = new RawRsa(pubKey, privKey);
+                        prime1 = RawRsa.ToBigEndian(rr.P);
+                        prime2 = RawRsa.ToBigEndian(rr.Q);
+                    }
+                    var rsaParams = (RsaParms)pubKey.parameters;
+                    var exponent = rsaParams.exponent != 0
+                                            ? Globs.HostToNet(rsaParams.exponent)
+                                            : RsaParms.DefaultExponent;
+                    var modulus = (pubKey.unique as Tpm2bPublicKeyRsa).buffer;
+                    var alg = new BCryptAlgorithm(Native.BCRYPT_RSA_ALGORITHM);
+                    cs.Key = alg.LoadRSAKey(exponent, modulus, prime1, prime2);
+                    alg.Close();
+                    break;
+                }
+                case TpmAlgId.Ecc:
+                {
+                    var eccParms = (EccParms)pubKey.parameters;
+                    var eccPub = (EccPoint)pubKey.unique;
+                    var algId = RawEccKey.GetEccAlg(pubKey);
+                    if (algId == null)
+                    {
+                        return null;
+                    }
+                    bool isEcdsa = eccParms.scheme.GetUnionSelector() == TpmAlgId.Ecdsa;
+                    byte[] keyBlob = RawEccKey.GetKeyBlob(eccPub.x, eccPub.y, keyAlgId,
+                                                            !isEcdsa, eccParms.curveID);
+                    var alg = new BCryptAlgorithm(algId);
+                    cs.Key = alg.ImportKeyPair(Native.BCRYPT_ECCPUBLIC_BLOB, keyBlob);
+                    alg.Close();
+                    if (cs.Key == UIntPtr.Zero)
+                    {
+                        Globs.Throw("Failed to create new RSA key");
+                        return null;
+                    }
+                    break;
+                }
+                default:
+                    Globs.Throw<ArgumentException>("Algorithm not supported");
+                    cs = null;
+                    break;
+            }
+            return cs;
+        }
+
+        public byte[] Export(string bcryptBlobType)
+        {
+            //RSAParameters parms = RsaProvider.ExportParameters(bcryptBlobType == Native.BCRYPT_RSAPRIVATE_BLOB);
+            //var alg = new BCryptAlgorithm(Native.BCRYPT_RSA_ALGORITHM);
+            //var Key = alg.LoadRSAKey(parms.Exponent, parms.Modulus, parms.P, parms.Q);

```

Comment:
Remove these comments too please, especially since RsaProvider no longer exists, now it only serves to confuse.

Reply:
Fixed, thanks for noticing these!